File: app/src/main/java/com/example/aichat/ThemedActivity.java
package com.example.aichat;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public abstract class ThemedActivity extends AppCompatActivity {
    private int appliedThemeResId;

    @Override
    protected void attachBaseContext(Context newBase) {
        Context wrapped = ThemeSettingsHelper.wrapForFontScale(newBase);
        super.attachBaseContext(wrapped != null ? wrapped : newBase);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        appliedThemeResId = ThemeSettingsHelper.resolveThemeResId(this);
        setTheme(appliedThemeResId);
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onResume() {
        super.onResume();
        int currentThemeResId = ThemeSettingsHelper.resolveThemeResId(this);
        if (currentThemeResId != appliedThemeResId) {
            recreate();
        }
    }
}

File: app/src/main/java/com/example/aichat/ThemedActivity.kt
package com.example.aichat

import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

abstract class ThemedActivity : AppCompatActivity() {
    private var appliedThemeResId: Int = 0

    override fun attachBaseContext(newBase: Context?) {
        val wrapped = newBase?.let { ThemeSettingsHelper.wrapForFontScale(it) }
        super.attachBaseContext(wrapped ?: newBase)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        appliedThemeResId = ThemeSettingsHelper.resolveThemeResId(this)
        setTheme(appliedThemeResId)
        super.onCreate(savedInstanceState)
    }

    override fun onResume() {
        super.onResume()
        val currentThemeResId = ThemeSettingsHelper.resolveThemeResId(this)
        if (currentThemeResId != appliedThemeResId) {
            recreate()
        }
    }
}
